src/interpreter/plugin/DateTimePlugin.ts

```diff
@@ -491,6 +542,253 @@ export class DatePlugin extends FunctionPlugin {
+          + ((second   > 0) ? second   + 'S' : '')
+      }
+    )
+  }
+
+  public networkdays(ast: ProcedureAst, formulaAddress: SimpleCellAddress): InternalScalarValue {
+    return this.runFunction(ast.args, formulaAddress, this.metadata('NETWORKDAYS'),
+      (start, end, holidays) => this.networkdayscore(start, end, 1, holidays)
+      )
+  }
+
+  public networkdaysintl(ast: ProcedureAst, formulaAddress: SimpleCellAddress): InternalScalarValue {
+    return this.runFunction(ast.args, formulaAddress, this.metadata('NETWORKDAYS.INTL'),
+      (start, end, weekend, holidays) => this.networkdayscore(start, end, weekend, holidays)
+      )
+  }
+
+  public workday(ast: ProcedureAst, formulaAddress: SimpleCellAddress): InternalScalarValue {
+    return this.runFunction(ast.args, formulaAddress, this.metadata('WORKDAY'),
+      (start, end, holidays) => this.workdaycore(start, end, 1, holidays)
+    )
+  }
+
+  public workdayintl(ast: ProcedureAst, formulaAddress: SimpleCellAddress): InternalScalarValue {
+    return this.runFunction(ast.args, formulaAddress, this.metadata('WORKDAY.INTL'),
+      (start, end, weekend, holidays) => this.workdaycore(start, end, weekend, holidays)
+    )
+  }
+
+  private networkdayscore(start: number, end: number, weekend: InternalNoErrorCellValue, holidays?: SimpleRangeValue): InternalScalarValue {
+    start = Math.trunc(start)
+    end = Math.trunc(end)
+    if(start>end) {
+      [start, end] = [end, start]
+    }
+    if(typeof weekend !== 'number' && typeof weekend !== 'string') {
+      return new CellError(ErrorType.VALUE, ErrorMessage.WrongType)
+    }
+
+    let weekendPattern: Maybe<string>
+
+    if(typeof weekend === 'string') {
+      if(weekend.length !== 7 || !/^(0|1)*$/.test(weekend)) {
+        return new CellError(ErrorType.NUM, ErrorMessage.WeekendString)
+      } else {
+        weekendPattern = weekend
+      }
+    } else {
+      weekendPattern = workdayPatterns.get(weekend)
+      if(weekendPattern === undefined) {
+        return new CellError(ErrorType.NUM, ErrorMessage.BadMode)
+      }
+    }
+
+    const uniqueHolidays = (holidays !== undefined) ? simpleRangeToUniqueNumbers(holidays) : []
+    if(uniqueHolidays instanceof CellError) {
+      return uniqueHolidays
+    }
+
+    const filteredHolidays = uniqueHolidays.filter((arg) => {
+      const val = this.interpreter.dateHelper.relativeNumberToAbsoluteNumber(arg)
+      const i = (val-1)%7
+      return (weekendPattern!.charAt(i) === '0')
+    })
+
+    return this.countWorkdays(start, end, weekendPattern, filteredHolidays)
+  }
+
+  private workdaycore(start: number, delta: number, weekend: InternalNoErrorCellValue, holidays?: SimpleRangeValue): InternalScalarValue {
+    start = Math.trunc(start)
+    delta = Math.trunc(delta)
+
+    if(typeof weekend !== 'number' && typeof weekend !== 'string') {
+      return new CellError(ErrorType.VALUE, ErrorMessage.WrongType)
+    }
+
+    let weekendPattern: Maybe<string>
```

Comment:
Wouldn't you prefer to have the `weekend->weekendPattern` mapping as a separate function? If I'm not mistaken we have a copy-pate here.